projects/Mallard/src/helpers/external-link.ts

```diff
@@ -0,0 +1,29 @@
+import { useNavigation } from '@react-navigation/native';
+import { useCallback } from 'react';
+import { NativeModules, Platform } from 'react-native';
+import { RouteNames } from 'src/navigation/NavigationModels';
+import { iosMajorVersion } from './platform';
+
+const externalLinkCanOpen = (): boolean => {
+	return Platform.OS === 'ios' && iosMajorVersion >= 16
+		? NativeModules.RNExternalLink.canOpen()
+		: false;
+};
+
+const externalLinkOpen = (): void => NativeModules.RNExternalLink.open();
+
+export const useExternalLink = () => {
+	const { navigate } = useNavigation();
+	const canOpen = externalLinkCanOpen();
+	const openExternalLink = useCallback(() => {
+		if (canOpen) {
```

Comment:
looking at Apple's docs, I think we also need to call StoreKit's `canMakePayments` first, in addition to `canOpen` before showing the modal (https://developer.apple.com/support/reader-apps/)